File: 2849_determine_if_a_cell_is_reachable_at_a_given_time/Solution.cs
// LeetCode 2849 - Determine if a Cell Is Reachable at a Given Time
// https://leetcode.com/problems/determine-if-a-cell-is-reachable-at-a-given-time/

using System;

public class Solution {
    public bool IsReachableAtTime(int sx, int sy, int fx, int fy, int t) {
        int need = Math.Max(Math.Abs(sx - fx), Math.Abs(sy - fy));
        if (need == 0) return t != 1;
        return t >= need;
    }
}

File: 2849_determine_if_a_cell_is_reachable_at_a_given_time/Solution.swift
// LeetCode 2849 - Determine if a Cell Is Reachable at a Given Time
// https://leetcode.com/problems/determine-if-a-cell-is-reachable-at-a-given-time/

class Solution {
    func isReachableAtTime(_ sx: Int, _ sy: Int, _ fx: Int, _ fy: Int, _ t: Int) -> Bool {
        let need = max(abs(sx - fx), abs(sy - fy))
        if need == 0 { return t != 1 }
        return t >= need
    }
}
